test_run.lua

```diff
@@ -169,6 +169,19 @@ local function wait_fullmesh(self, servers)
                 while true do
                     local cmd = wait_fullmesh_cmd:format(server_id)
                     local info = self:eval(server2, cmd)[1]
+                    if info ~= nil and info.upstream ~= nil then
+                        if info.upstream.status == 'stopped' and
+                                info.upstream.message ~= nil then
+                            log.info("fullmesh failed to connect %s -> %s",
+                                     ": status %s : message %s",
+                                     server2, server, info.upstream.status,
+                                     info.upstream.message)
+                            return false
+                        end
+                        log.info("connecting %s -> %s : status %s : message %s",
+                                 server2, server, info.upstream.status,
+                                 info.upstream.message)
```

Comment:
I prefer to don't lean on the fact that `nil` is okay for %s in log.info() (I doubt that it is documented and cannot be changed once) and use tostring() for status and message both.

Reply:
Added tostring routines, removed check of nil value for box.info() routine.